packages/common/src/utils/map.ts

```diff
@@ -1,11 +1,16 @@
+const isDefaultValueAFunction = <V>(value: unknown): value is () => V => typeof value === 'function';
+
+export function getOrDefault<K, V>(map: Map<K, V>, key: K, defaultValue: V): V;
+export function getOrDefault<K, V>(map: Map<K, V>, key: K, getDefaultValue: () => V): V;
 /**
  * Retrieve the value at the given key inside the map.
  * If the key does not exist, insert the default value into the map and return that value.
  */
-export const getOrDefault = <K, V>(map: Map<K, V>, key: K, defaultValue: V): V => {
+export function getOrDefault<K, V>(map: Map<K, V>, key: K, defaultValue: V | (() => V)): V {
```

Comment:
this is also the only way to make the default anything that's pass by reference (ex. object or array)